sdk/azcore/core.go

```diff
@@ -22,6 +22,24 @@ type AccessToken = exported.AccessToken
 // TokenCredential represents a credential capable of providing an OAuth token.
 type TokenCredential = exported.TokenCredential
 
+// KeyCredential contains an authentication key used to authenticate to an Azure service.
+type KeyCredential = exported.KeyCredential
+
+// NewKeyCredential creates a new instance of [KeyCredential] with the specified values.
+//   - key is the authentication key
+func NewKeyCredential(key string) (*KeyCredential, error) {
```

Comment:
Should these have options?

Reply:
Famous last words but I think we'll be okay without options because this is just a container for a string. The policy does all the interesting stuff.

Reply:
I agree and kind of hate having to add two more options placeholder types (this is one place where variadic config has the edge).  I just want to make sure we're all in agreement.

Reply:
Normally I want them but this is such a limited case...

Reply:
Since the only parameter is a key whose format/values is defined by the service, the client cannot validate it and therefore I think returning an error is unnecessary.

Reply:
The only reason it returns an error at present is if the key is the empty string.

Will remove per discussion.